private/pkg/command/process.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"context"
+	"os/exec"
+
+	"go.uber.org/multierr"
+)
+
+type process struct {
+	cmd          *exec.Cmd
+	doneCallback func()
+	wait         chan error
```

Comment:
This is an unbuffered channel, it will block unless there is a receiver already installed, meaning that the call in `monitor` will block unless someone has called wait. This has the same issue as  before - a potential deadlock unless the user calls `Wait`. This needs to be a buffered channel.

Reply:
Whoops, that came from a refactor where I passed error into the callback at first (doesn't seem useful, also doesn't encourage actually calling Wait).

I'll capture err and callback first. Sticks with style better. Also I'll add testing to actually cover the complete behavior of this.

Reply:
The channel is still unbuffered - the  implementation as it currently is will still leave zombie goroutines